src/app/container/register-tool/register-tool.component.html

```diff
@@ -378,16 +379,23 @@ <h4 mat-dialog-title>Register Tool</h4>
               [(ngModel)]="hostedTool.path"
               minlength="3"
               maxlength="128"
-              [pattern]="hostedTool.registryProvider === 'Amazon ECR' ? validationPatterns.repoNameWithSlashesImagePath : validationPatterns.imagePath"
+              [pattern]="
+                hostedTool.registryProvider === 'Amazon ECR'
+                  ? validationPatterns.repoNameWithSlashesImagePath
+                  : validationPatterns.imagePath
+              "
               required
               placeholder="e.g. cancercollaboratory/dockstore-tool-liftover"
               matTooltip="Docker Image Repository"
             />
-            <mat-hint *ngIf="hostedTool.registryProvider === 'Amazon ECR'">Has the form &lt;namespace&gt;/&lt;name&gt;. An empty namespace, represented by '_', must be used if the registry is private.</mat-hint>
-            <mat-hint *ngIf="hostedTool.registryProvider !== 'Amazon ECR'">Has the form &lt;namespace&gt;/&lt;name&gt;</mat-hint>
+            <mat-hint *ngIf="hostedTool.registryProvider === 'Amazon ECR'"
+              >Has the form &lt;namespace&gt;/&lt;name&gt;. If the registry is private, use '_' as the namespace.</mat-hint
```

Comment:
I think this message is very clear, and maybe you could use it or a variant for the error message (see my other comment).